### ISSUE TITLE
Add SuppressLinkerWarningAttribute

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#35339]
This is in support of mono/linker#988 - see the spec there for motivation.

We need a way to selectively suppress warnings about linker-unfriendly patterns.

This attribute is largely inspired by SuppressMessageAttribute (that unfortunately cannot be used because it's marked `[Conditional]` under a condition that we wouldn't expect to enable when the linker runs).

```csharp
namespace System.Runtime.CompilerServices
{
    [AttributeUsage(
     AttributeTargets.All,
     Inherited = false,
     AllowMultiple = true
     )
    ]
    public sealed class SuppressLinkerWarningAttribute : Attribute
    {
        public SuppressLinkerWarningAttribute(string category, string checkId)
        {
            Category = category;
            CheckId = checkId;
        }

        public string Category { get; }
        public string CheckId { get; }
        public string? MessageId { get; set; }
        public string? Justification { get; set; }
    }
}
```

**Comment by MichalStrehovsky** [comment]
Cc @vitek-karas

**Comment by terrajobst** [comment]
* We should position it as a peer to `SuppressMessageAttribute`, `UnconditionalSuppressMessageAttribute`
* It should be `System.Diagnostics.CodeAnalysis`
* The behavior is the same, except it's not marked `[Conditonal]` and thus will always remain in metadata 
* We considered establishing an inheritance relationshop with `SuppressMessageAttribute` but concluded we don't need it (neither for docs nor for consumption)

```C#
namespace System.Diagnostics.CodeAnalysis
{
    [AttributeUsage(AttributeTargets.All, Inherited = false, AllowMultiple = true)]
    public sealed class UnconditionalSuppressMessageAttribute : Attribute
    {
    	// Same API as SuppressMessageAttribute
    }
}
```

**Comment by eerhardt** [comment]
Closed in #35387.
